.github/workflows/pytest_and_coverage.yml

```diff
@@ -0,0 +1,46 @@
+name: PyTest and Coverage
+
+on: ["push", "pull_request", "workflow_dispatch"]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # supported python versions can be found here
+        # https://github.com/actions/python-versions/releases
+        #
+        # Please bump to the latest unreleased candidate
+        # when you come across this and have a moment to spare!
+        python-version: ["3.10", "3.11", "3.12", "3.13"]
+    steps:
+    - uses: actions/checkout@v5
+    - name: Install uv
+      uses: astral-sh/setup-uv@v6
+      with:
+        version: "0.8.18"
+        python-version: ${{ matrix.python-version }}
+    - name: Install the project
+      run: uv sync --all-extras --dev
+      # run: uv sync --locked --all-extras --dev  # consider using --locked once in the repo
+
+    - name: Run tests with coverage
+      run: >
+        uv run pytest
+        --cov-report html:${{ github.workspace }}/coverage_report_html_${{ matrix.python-version }}
+        --cov-report xml:${{ github.workspace }}/coverage_report_xml_${{ matrix.python-version }}.xml
+        --cov-report markdown-append:$GITHUB_STEP_SUMMARY
+        --cov-config .coveragerc
+        --cov marshmallow_jsonschema
+    
+    - name: Archive coverage
```

Comment:
I think this archiving stuff can be combined. Will do this in #33 though after the builds are more organised.